components/brave_sync/time_limited_words.cc

```diff
@@ -210,4 +212,35 @@ std::string TimeLimitedWords::GenerateResultToText(
   }
 }
 
+base::Time TimeLimitedWords::GetNotAfter(
+    const std::string& time_limited_words) {
+  std::vector<std::string> words = base::SplitString(
+      time_limited_words, " ", base::WhitespaceHandling::TRIM_WHITESPACE,
+      base::SplitResult::SPLIT_WANT_NONEMPTY);
+  size_t num_words = words.size();
+
+  if (num_words != kWordsV2Count) {
+    return base::Time();
+  }
+
+  int days_encoded = GetIndexByWord(words[kWordsV2Count - 1]);
+  const base::Time anchorTime = GetWordsV2Epoch() + base::Days(days_encoded);
+
+  // Iterate with 1 hour interval to find the point when the words will be
+  // rejected
+  for (base::Time time_to_check = anchorTime;;
+       time_to_check += base::Hours(1)) {
```

Comment:
This loop is no better than the original version.
I think it would be easy to understand and clear to use same `GetRoundedDaysDiff` with original logic rather than hard coded 36hr which needs further explanation.

Reply:
Probably something like `DCHECK_EQ(GetRoundedDaysDiff(anchor_time + base::Hours(36), anchor_time), 2));`
and declare 36h as constant should be enough.

Reply:
I reverted the looped version and tried to put the best description I could make. DCHECKs are also there.

https://github.com/brave/brave-core/pull/23051/commits/b03e68db97fe0fe622952b80d41e00062ee4046c

thanks for the advises, tips and discussion :+1: